File: [108] 将有序数组转换为二叉搜索树.cpp
#include <vector>
using namespace std;

struct TreeNode {
	int val;
	TreeNode* left;
	TreeNode* right;
	TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
	TreeNode* sortedArrayToBST(vector<int>& nums) {
		return dp(nums, 0, nums.size());
	}
	TreeNode* dp(vector<int>& nums, int left, int right) {
		if (left == right)
			return NULL;

		int mid = (left + right) / 2;
		TreeNode* root = new TreeNode(nums[mid]);
		root->left = dp(nums, left, mid);
		root->right = dp(nums, mid + 1, right);
		return root;
	}
};

File: solutions/[108]将有序数组转换为二叉搜索树.py
# Definition for a binary tree node.
from typing import List


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def sortedArrayToBST(self, nums: List[int]) -> TreeNode:
        def helper(l, r):
            if l > r:
                return None
            p_root = (l+r) // 2

            return TreeNode(
                nums[p_root],
                helper(l, p_root-1),
                helper(p_root+1, r)
            )
        return helper(0, len(nums)-1)